src/include/ikarus/finiteElements/mechanics/enhancedAssumedStrains.hh

```diff
@@ -0,0 +1,406 @@
+    Eigen::Matrix3d T0InverseTransformed;
+  };
+
+  template <typename Geometry>
+  struct EASH1E9 {
+    static constexpr int strainSize         = 6;
+    static constexpr int enhancedStrainSize = 9;
+
+    EASH1E9() = default;
+    explicit EASH1E9(const Geometry& geometry)
+        : geometry{std::make_unique<Geometry>(geometry)}, T0InverseTransformed{calcTransformationMatrix3D(geometry)} {}
+
+    auto calcM(const Dune::FieldVector<double, 3>& quadPos) const {
+      Eigen::Matrix<double, strainSize, enhancedStrainSize> M;
+      M.setZero();
+      const double xi   = quadPos[0];
+      const double eta  = quadPos[1];
+      const double zeta = quadPos[2];
+      M(0, 0)           = xi - 0.5;
+      M(1, 1)           = eta - 0.5;
+      M(2, 2)           = zeta - 0.5;
+      M(3, 3)           = xi - 0.5;
+      M(3, 4)           = eta - 0.5;
+      M(4, 5)           = xi - 0.5;
+      M(4, 6)           = zeta - 0.5;
+      M(5, 7)           = eta - 0.5;
+      M(5, 8)           = zeta - 0.5;
+      const double detJ = geometry->integrationElement(quadPos);
+      M                 = T0InverseTransformed / detJ * M;
+      return M;
+    }
+    std::unique_ptr<Geometry> geometry;
+    Eigen::Matrix<double, 6, 6> T0InverseTransformed;
+  };
+
+  template <typename Geometry>
+  struct EASH1E21 {
+    static constexpr int strainSize         = 6;
+    static constexpr int enhancedStrainSize = 21;
+
+    EASH1E21() = default;
+    explicit EASH1E21(const Geometry& geometry)
+        : geometry{std::make_unique<Geometry>(geometry)}, T0InverseTransformed{calcTransformationMatrix3D(geometry)} {}
+
+    auto calcM(const Dune::FieldVector<double, 3>& quadPos) const {
+      Eigen::Matrix<double, strainSize, enhancedStrainSize> M;
+      M.setZero();
+      const double xi   = quadPos[0];
+      const double eta  = quadPos[1];
+      const double zeta = quadPos[2];
+      M(0, 0)           = xi - 0.5;
+      M(1, 1)           = eta - 0.5;
+      M(2, 2)           = zeta - 0.5;
+      M(3, 3)           = xi - 0.5;
+      M(3, 4)           = eta - 0.5;
+      M(4, 5)           = xi - 0.5;
+      M(4, 6)           = zeta - 0.5;
+      M(5, 7)           = eta - 0.5;
+      M(5, 8)           = zeta - 0.5;
+
+      M(3, 9)  = xi* zeta - 0.25;
+      M(3, 10) = eta* zeta - 0.25;
+      M(4, 11) = xi* eta - 0.25;
+      M(4, 12) = eta* zeta - 0.25;
+      M(5, 13) = xi* eta - 0.25;
+      M(5, 14) = xi* zeta - 0.25;
+
+      M(0, 15) = xi* eta - 0.25;
+      M(0, 16) = xi* zeta - 0.25;
+      M(1, 17) = xi* eta - 0.25;
+      M(1, 18) = eta* zeta - 0.25;
+      M(2, 19) = xi* zeta - 0.25;
+      M(2, 20) = eta* zeta - 0.25;
+
+      const double detJ = geometry->integrationElement(quadPos);
+      M                 = T0InverseTransformed / detJ * M;
+      return M;
+    }
+
+    std::unique_ptr<Geometry> geometry;
+    Eigen::Matrix<double, 6, 6> T0InverseTransformed;
+  };
+
+  /// ***** For a new EAS formulation, add a struct here *****
```

Comment:
All this EAS comments easily get out-of date. 
I would propose that we really clean this and start writing a small eas chapter in the documentation.